kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/builders.kt

```diff
@@ -27,42 +25,35 @@ fun <A> arbitrary(edgecases: List<A>, fn: (RandomSource) -> A): Arb<A> = object
  * generates values from the given function.
  */
 fun <A> arbitrary(edgecases: List<A>, shrinker: Shrinker<A>, fn: (RandomSource) -> A): Arb<A> = object : Arb<A>() {
-   override fun edgecases(): List<A> = edgecases
+   override fun edgecase(rs: RandomSource): A? = if (edgecases.isEmpty()) null else edgecases.random(rs.random)
    override fun sample(rs: RandomSource): Sample<A> = sampleOf(fn(rs), shrinker)
    override fun values(rs: RandomSource): Sequence<Sample<A>> = generateSequence { sampleOf(fn(rs), shrinker) }
 }
 
 /**
- * Creates a new [Arb] using the given edgecase generator and generates samples from the given sample generator.
+ * Creates a new [Arb] that generates edgecases from the given [edgecaseFn] function
+ * and generates samples from the given [sampleFn] function.
  */
-internal fun <A> arbitrary(edgecaseGenerator: EdgeConfig.(RandomSource) -> A, sampleGenerator: (RandomSource) -> A): Arb<A> =
+internal fun <A> arbitrary(edgecaseFn: (RandomSource) -> A?, sampleFn: (RandomSource) -> A): Arb<A> =
```

Comment:
Now that we've decided to remove list edgecases, would this still be an internal fun or do we make this public? I think this is good to be internal but just wanted to get your opinion

Reply:
I think without this builder it's impossible to combine some arbs where you want to randomly choose from constituent arbs.
I will make it public.